.github/workflows/validate-gha-refs.yml

```diff
@@ -0,0 +1,68 @@
+name: Validate GHA References
+
+on:
+  workflow_call:
+    inputs:
+      path-patterns:
+        required: false
+        type: string
+        default: ".github"
+        description: A comma separated list of path patterns (relative to repository root) in which to check for GHA files
+      include-patterns:
+        required: false
+        type: string
+        default: "*.yml,*.yaml"
+        description: A comma separated list of include patterns to select GHA files for checking
+      repo-patterns:
+        required: false
+        type: string
+        default: ${{ github.repository }}
+        description: A comma separated list of repository patterns to consider when evaluating reusable workflow and action references
+      expected-ref:
+        required: false
+        type: string
+        default: ${{ github.ref_name }}
+        description: The expected reference value to be used by the reusable workflows and actions 
+jobs:
+  check-workflows:
+    name: Check Workflows/Actions
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout Repository
+        uses: Lombiq/GitHub-Actions/.github/actions/checkout@dev
+
+      - name: Find Ref Mismatches
+        shell: pwsh
+        run: |
+          $repoPatterns = "${{ inputs.repo-patterns }}".Split(",")
+          for($i = 0; $i -lt $repoPatterns.Count; $i++)  { $repoPatterns[$i] = "uses:\s*" + $repoPatterns[$i]; }          
+
+          $mismatches = Get-ChildItem -Path ${{ inputs.path-patterns }} -Include ${{ inputs.include-patterns }} -Force -Recurse | 
+          Select-String -Pattern $repoPatterns | 
+          Select-String -Pattern '@${{ inputs.expected-ref }}$' -NotMatch
+          
+          if ($mismatches.Count -gt 0)
+          {
+            "Please update the following lines to match the expected GitHub Actions reference: '${{ inputs.expected-ref }}'" >> $env:GITHUB_STEP_SUMMARY
+
+            for($i = 0; $i -lt $mismatches.Count; $i++) 
```

Comment:
We scratched having the workflow update all the references automatically (for now) because of the following I think
>Automatically managing these would necessitate commits to the repos (including superprojects and submodules). This looks like an issue with submodules.

And if I can add a step to the [validate-this-gha-refs](https://github.com/Lombiq/GitHub-Actions/blob/issue/OSOE-517/.github/workflows/validate-this-gha-refs.yml) workflow to consider only those files that have changed in the commit, then I think we can avoid unnecessarily changing GHA refs.

Reply:
I see, okay thank you!